pyrsistent/_pmap.py

```diff
@@ -125,7 +125,27 @@ def __len__(self):
     def __repr__(self):
         return 'pmap({0})'.format(str(dict(self)))
 
-    __eq__ = Mapping.__eq__
+    def __eq__(self, other):
+        if self is other:
+            return True
+        if not isinstance(other, Mapping):
+            return NotImplemented
+        if len(self) != len(other):
+            return False
+        if isinstance(other, PMap):
+            try:
+                if hash(self) != hash(other):
+                    return False
+            except TypeError:
+                # self or other contains non-hashable values
+                pass
+            if self._buckets == other._buckets:
+                return True
+            return dict(self.iteritems()) == dict(other.iteritems())
```

Comment:
> I think, the worst case would rather be two equal pmaps with the same number of buckets, but elements inserted into the same bucket in a different order.

The only way I see to address this would be to model a bucket using a dict instead of a list, so the _buckets equality check would tell any two pmaps with the same number of buckets apart. But this would quite possibly hurt pmap creation and insertion performance.